website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -1,182 +1,315 @@
+```
 
-## Configuring your test environment
+To update `config/rbac/role.yaml` after changing the markers, run `make manifests`.  Also, you might want to configure additional permissions for your project, see [RBAC markers][rbac_markers] doc. 
 
-Projects are scaffolded with unit tests that utilize the [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest)
-library, which requires certain Kubernetes server binaries be present locally.
-Installation instructions can be found [here][env-test-setup].
+By default, the projects are `cluster-scoped`. Check that your role is of the `ClusterRole` Kind. For further information check the [operator-scope][operator-scope] doc. 
 
-## Operator Manifests
+**Note** You  would like to check the final result of our [memcached_controller.go][memcached_controller] migrated to the new layout. 
 
-### Operator deployment manifests
+## Migrating the Main.go
 
-The operator deployment manifest [`deploy/operator.yaml`][deployment_yaml] from the old project should be copied over to `config/manager/manager.yaml` in the new project.
+If you are using third-party API's in your project for example then, it means that you need to copy your customizations 
+from `cmd/manager/main.go` to `main.go`. You’ll also need to ensure all of the needed schemes have been registered.
 
-**Note**:
-The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the `deploy/operator.yaml` manifest of SDK project.
-> **// TODO:** Explain label propagation for metrics collection.
+By checking our new `main.go` we will find that:
 
-### RBAC permissions
+- The SDK `leader.Become` was replaced by the [leaderElection][leaderElection]. However, we still able to use the 
+previous one as well by adding the [operator-lib][operator-lib] as dependency fo the project.
+- The default port used by the metric endpoint binds to from `:8383` to `:8080`. To continue using port `8383`, specify `--metrics-addr=:8383` when you start the operator. 
 
-The RBAC manifests present in `config/rbac/` of a Kubebuilder project are generated through the [RBAC markers][rbac_markers] present in `controller/<kind>_controller.go`. They can be added as comments above the `Reconcile()` method.
+## Configuring your test environment
 
-In our example of memcached operator, add the following RBAC markers in `memcached_controller.go`:
+Projects are scaffolded with tests that requires certain Kubernetes server binaries be present locally. In this way, we'll need to setup it locally by running:
 
-```Go
+```sh
+curl -sSLo setup_envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/kubebuilder/master/scripts/setup_envtest_bins.sh 
+chmod +x setup_envtest.sh
+eval $(./setup_envtest.sh v1.18.2 v3.4.3)
+```
 
-...
-// +kubebuilder:rbac:groups=apps,resources=deployments;pods;daemonsets;replicasets;statefulsets,verbs=get;update;patch;list;create;delete;watch
+**Note:** More info can be found [here][env-test-setup].
```

Comment:
This section needs to be updated based on the discussion in the Kubebuilder meeting.